src/onepasswordconnectsdk/async_client.py

```diff
@@ -0,0 +1,371 @@
+
+from onepasswordconnectsdk.connect import PathBuilder
+from onepasswordconnectsdk.serializer import Serializer
+from onepasswordconnectsdk.utils import build_headers, is_valid_uuid
+from onepasswordconnectsdk.errors import (
+    FailedToRetrieveItemException,
+    FailedToRetrieveVaultException,
+)
+from onepasswordconnectsdk.models import Item, ItemVault
+
+
+class AsyncClient:
+    """Python Async Client Class"""
+
+    def __init__(self, url: str, token: str):
+        """Initialize async client"""
+        self.url = url
+        self.token = token
+        self.session = self.create_session(url, token)
+        self.serializer = Serializer()
+
+    def create_session(self, url: str, token: str):
+        return httpx.AsyncClient(base_url=url, headers=self.build_headers(token))
+
+    def build_headers(self, token: str):
+        return build_headers(token)
+
+    def __del__(self):
+        self.session.aclose()
+
+    async def get_file(self, file_id: str, item_id: str, vault_id: str):
+        url = PathBuilder().vaults(vault_id).items(item_id).files(file_id).build()
+        response = await self.build_request("GET", url)
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            raise FailedToRetrieveItemException(
+                f"Unable to retrieve item. Received {response.status_code}\
+                                     for {url} with message: {response.json().get('message')}"
+            )
+        return self.serializer.deserialize(response.content, "File")
+
+    async def get_files(self, item_id: str, vault_id: str):
+        url = PathBuilder().vaults(vault_id).items(item_id).files().build()
+        response = await self.build_request("GET", url)
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            raise FailedToRetrieveItemException(
+                f"Unable to retrieve item. Received {response.status_code}\
+                             for {url} with message: {response.json().get('message')}"
+            )
+        return self.serializer.deserialize(response.content, "list[File]")
+
+    async def get_file_content(self, file_id: str, item_id: str, vault_id: str, content_path: str = None):
+        url = content_path
+        if content_path is None:
+            url = PathBuilder().vaults(vault_id).items(item_id).files(file_id).content().build()
+        response = await self.build_request("GET", url)
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            raise FailedToRetrieveItemException(
+                f"Unable to retrieve items. Received {response.status_code} \
+                     for {url} with message: {response.json().get('message')}"
+            )
+        return response.content
+
+    async def download_file(self, file_id: str, item_id: str, vault_id: str, path: str):
+        file_object = await self.get_file(file_id, item_id, vault_id)
+        filename = file_object.name
```

Comment:
`[non-blocking]`
I see that the sync client does this (`client.py` - L83):
```py
        filename = file_object.name or "1password_item_file.txt"	
```

Should we match this for the async as well for feature parity? 🤔

Reply:
Agree, that should be the same. Added it here